Client/Frontend/Settings/SettingsTableViewController.swift

```diff
@@ -147,14 +147,14 @@ private class PaddedSwitch: UIView {
 // A helper class for settings with a UISwitch.
 // Takes and optional settingsDidChange callback and status text.
 class BoolSetting: Setting {
-    let prefKey: String
+    let prefKey: String? // Sometimes a superclass will manage its own pref setting. In that case the prefkey will be nil
 
     fileprivate let prefs: Prefs
     fileprivate let defaultValue: Bool
-    fileprivate let settingDidChange: ((Bool) -> Void)?
+    var settingDidChange: ((Bool) -> Void)?
```

Comment:
The completely undocumented intention of `settingDidChange` was so that leave classes could add their own behaviours without subclassing.

I've suggested a way of doing this below.